src/app/kuma/components/ApplicationShell.vue

```diff
@@ -26,8 +26,9 @@
               v-slot="{ data }"
               :src="`/control-plane/version/latest`"
             >
+              <!-- make sure we have data but don't show errors or loaders -->
               <KAlert
-                v-if="env('KUMA_VERSION') !== data?.version"
+                v-if="data && env('KUMA_VERSION') !== data.version"
```

Comment:
This is the fix, when data is `undefined` it doesn't equal `KUMA_VERSION`, which means the upgrade alert can show for a sort period (and every time you refresh the browser)